plan/config.go

```diff
@@ -104,10 +114,54 @@ func parseBehavior(d string) Behavior {
 		ClientAxis: clientAxis,
 		ParamsAxes: values,
 	}
-
 	return behavior
 }
 
+func validateAndApplyFilters(filterMap map[string][]Filter, behaviors Behaviors) (Behaviors, error) {
+	for i := range behaviors {
+		behavior := &behaviors[i]
+		filters := filterMap[behavior.Name]
+		for _, filter := range filters {
+			for match := range filter.AxisMatches {
+				if match != behavior.ClientAxis && !axisContainsRule(match, behavior.ParamsAxes) {
+					return nil, fmt.Errorf("%v is not defined in axis for %v", match, behavior.Name)
+				}
+			}
+		}
+		behavior.SkipFilters = filters
+	}
+	return behaviors, nil
+}
+
+func axisContainsRule(item string, axis []string) bool {
```

Comment:
Axis - Singular, Axes - plural and updated the function accordingly. Also this function now becomes part of the entities.go ( on behavior)